server/src/watchdog.js

```diff
@@ -138,3 +149,45 @@ exports.submit = function(shot) {
     mozlog.error("watchdog-failed-submission", {err: e});
   });
 };
+
+exports.handleResult = function(req) {
+  getSubmission(req.params.submissionId).then(record => {
+    if (record.request_id !== req.body.watchdog_id) {
+      throw new Error(`Received mismatching Watchdog ID in callback for submission ${record.id}.`);
+    }
+    if (req.query.nonce !== record.nonce) {
+      throw new Error(`Received mismatching nonce in callback for submission ${record.id}.`);
+    }
+    if (record.positive_result !== null) {
+      throw new Error(`Watchdog submission ${record.id} already has a result.`);
+    }
+
+    if (req.body.positive === false) {
+      handleNegative(record);
+    } else {
+      handlePositive(record);
+    }
+  }).catch(e => {
+    mozlog.error("watchdog-failed-callback", {err: e});
```

Comment:
This should probably use `responses.sendError(req, res, e)`, or else the catch should happen in `server.js`. Or you should use `ravenclient.captureRavenException`